eZ/Publish/Core/Persistence/Cache/UrlAliasHandler.php

```diff
@@ -251,26 +251,33 @@ public function locationMoved($locationId, $oldParentId, $newParentId)
         );
 
         $return = $this->persistenceHandler->urlAliasHandler()->locationMoved($locationId, $oldParentId, $newParentId);
+        $this->cache->clear('urlAlias');
 
-        $this->cache->clear('urlAlias');//TIMBER! (Will have to load url aliases for location to be able to clear specific entries)
         return $return;
     }
 
     /**
      * @see eZ\Publish\SPI\Persistence\Content\UrlAlias\Handler::locationCopied
      */
-    public function locationCopied($locationId, $oldParentId, $newParentId)
+    public function locationCopied($locationId, $newLocationId, $newParentId)
     {
         $this->logger->logCall(
             __METHOD__,
             array(
-                'location' => $locationId,
-                'oldParent' => $oldParentId,
+                'oldLocation' => $locationId,
+                'newLocation' => $newLocationId,
                 'newParent' => $newParentId,
             )
         );
 
-        return $this->persistenceHandler->urlAliasHandler()->locationCopied($locationId, $oldParentId, $newParentId);
+        $return = $this->persistenceHandler->urlAliasHandler()->locationCopied(
+            $locationId,
+            $newLocationId,
+            $newParentId
+        );
+        $this->cache->clear('urlAlias');
```

Comment:
As mentioned, think this one can be set as `$this->cache->clear('urlAlias', 'url');` given we only care about possible negative matches when creating new entries.